src/main.rs

```diff
@@ -76,3 +68,18 @@ fn main() -> anyhow::Result<()> {
 
     Ok(()) as Result<(), anyhow::Error>
 }
+
+fn filter_from_verbosity(verbose_flag_count: u8) -> log::LevelFilter {
+    match verbose_flag_count {
+        0 => log::LevelFilter::Warn,
+        1 => log::LevelFilter::Info,
+        2 => log::LevelFilter::Debug,
+        _ => log::LevelFilter::Trace,
+    }
+}
```

Comment:
I'm not a huge fan of having the verbosity be a number that vaguely maps to a log level. What do you think about something similar to this instead?

```rust
#[derive(clap::Parser)]
struct Args {
    /// ...
    #[clap(arg_enum)]
    log_level: LogLevel,
}

#[derive(clap::ArgEnum, Clone)]
enum LogLevel {
    Debug,
    Info,
    Warning,
    Error,
}
```
That would allow you to specify the log level with
```
tricore-probe --log-level=trace
```
Which I think is semantically stronger and more idiomatic

Reply:
Thats a bit better, I like it!